src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -567,13 +569,21 @@ void InitializeCaches()
                 ErrorUtilities.ThrowInternalError("Only one project cache plugin may be set on the BuildManager during a begin / end build session");
             }
 
-            LogMessage(ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("LoadingProjectCachePlugin", pluginDescriptor.GetDetailedDescription()));
+            lock (_syncLock)
```

Comment:
I'm wondering if it's worth it to try to minimize the time spent in locks, especially this one. Something like:
```C#
if (_projectCacheService == null) {
  // var p = ProjectCacheService.FromDescriptorAsync(...);
  lock (_syncLock) {
    if (_projectCacheService == null) {
      _projectCacheService = p;
    }
  }
}